threat_model/threat_model.md

```diff
@@ -0,0 +1,70 @@
+# Terraform Vault Backend Threat Model
+
+## Attacker personas
+
+| Capabilities | Acronym |
+|--------------|---------|
+| Unauthenticated Internet attacker | UIA |
+| Local network attacker | LNA |
+| Malicious user | MUA |
+| Compromised TVB | CTA |
+| Compromised Vault | CVA |
+
+
+## Assumptions
+- Malicious TF state can cause information leaks when used by TF client to deploy an application.
+- TVB and Vault are deployed on a local network separated from the Internet.
```

Comment:
As this is a public (open source) project, there is no limitation or recommendation to keep it hidden behind a corp. firewall. Lets assume the opposite in or threat model (TVB and Vault are deployed on the Internet) and adjust exposure to `UIA` accordingly.  

We can use LAN deployment only as a Countermeasure, if there are no better options.

Reply:
I guess we just replace LNA with UIA then, and update exposure accordingly.